File: Recursion/MyPow/MyPow.go
package main

import "fmt"

func myPow(x float64, n int) float64 {
	ans := 1.0
	for n > 0 {
		if n%2 == 1 {
			ans = ans * x
			n = n - 1
		}
		x = x * x
		n = n / 2
	}
	return ans
}
func main() {
	fmt.Println(myPow(2.00000, 10))
}

File: Recursion/MyPow/java/MyPow.java
class MyPow{
    public static double f(double x, int n){
        if(n==0) return 1.0;
        if(n<0){
            x = 1/x;
            n = -1*n;
        }
        if(n%2!=0) return x * f(x, n-1);
        return f(x*x, n/2);
    }
    public static void main(String[] args){
        System.out.println("2^10 : "+f(2.000000, 10));
    }
}